namesys/namesys.go

```diff
@@ -86,16 +86,19 @@ func (ns *mpns) Resolve(ctx context.Context, name string, options ...opts.Resolv
 }
```

Comment:
Could someone _carefully_ check the changes I made here. I think they're right, but this is tricky code.